src/Simplify.elm

```diff
@@ -3914,6 +3957,64 @@ equalityChecks isEqual =
         ]
 
 
+compareWithZeroChecks :
+    OperatorApplicationCheckInfo
+    -> Bool
+    -> Node Expression
+    ->
+        Maybe
+            { message : String
+            , details : List String
+            , fnRange : Range
+            , replacement : String
+            }
+compareWithZeroChecks checkInfo isEqual node =
+    [ ( Fn.List.isEmpty, Fn.List.length, "List" )
+    , ( Fn.Dict.isEmpty, Fn.Dict.size, "Dict" )
+    , ( Fn.Set.isEmpty, Fn.Set.size, "Set" )
+
+    -- , ( Fn.String.isEmpty, Fn.String.length, "String" ) is this the best replacement? Should it be == ""?
+    ]
+        |> findMap
+            (\( newFn, oldFn, structName ) ->
+                case
+                    AstHelpers.getSpecificFnCall
+                        oldFn
+                        checkInfo.lookupTable
+                        node
+                of
+                    Just call ->
+                        let
+                            newName : String
+                            newName =
+                                qualifiedToString (qualify newFn checkInfo)
+
+                            replacement =
+                                if isEqual then
+                                    newName
+
+                                else
+                                    "(not " ++ newName ++ ")"
```

Comment:
this is wrong! `not : Bool -> Bool` but we want something like `invert : (a -> Bool) -> Bool -> Bool` which doesn't exist. So I'll need to do some work rearranging the parentheses.

Reply:
this was very easy to fix lol. `<<` ezpz